src/autoqasm/transpiler/transpiler.py

```diff
@@ -131,6 +137,7 @@ def transform_ast(
         node = assignments.transform(node, ctx)
         node = lists.transform(node, ctx)
         node = slices.transform(node, ctx)
+        node = typecast.transform(node, ctx)
```

Comment:
I think you'll want to call this _before_ the `assignments.transform` call. The reason is that you want any variable assignment to use the type _after_ it has been cast. This is so that `syndrome = int(measure[0,1])` produces an `int` variable in OpenQASM rather than a `bit`, as you're currently seeing.

Reply:
That makes a lot of sense! I moved it up so that it gets called before assignments. I think that this is the main thing that was confusing me. I added a few changes in the other files to match this change.